python/tk_multi_screeningroom/screeningroom.py

```diff
@@ -53,20 +56,32 @@ def _launch_rv(base_url, cmd, source=None, path_to_rv=None):
     if source:
         args.append(source)
 
+    combined_args = " " + " ".join(args)
     # We should encode the args so that no special characters exist in the command
-    encoded_args = (" " + " ".join(args)).encode("hex")
+    # Encode the string into binary so that we can convert the args to a hex
+    binary_str = six.ensure_binary(combined_args)
+    # Encode the binary into a hex
+    encoded_args = codecs.encode(binary_str, "hex")
+    # Now convert the binary back into a string which contains the hex representing the args.
+    hex_encoded_args_string = six.ensure_str(encoded_args)
 
     # If no path to RV was provided, us the SG site rvlink protocol to launch RV
     if not path_to_rv:
         # Encode the RV args. We'll use shotgun to redirect to the RV app via the rvlink
         # custom protocol
-        url = "%s/rvlink/baked/%s" % (base_url, encoded_args)
+        url = "%s/rvlink/baked/%s" % (base_url, hex_encoded_args_string)
         webbrowser.open(url)
     else:
-        url = "rvlink://baked/%s" % (encoded_args)
+        url = "rvlink://baked/%s" % (hex_encoded_args_string)
         cmdLine = " ".join(['"%s"' % path_to_rv, url])
         print("Running %s" % cmdLine)
         subprocess.Popen(cmdLine, shell=True)
+        # For some reason, on Python 3, it doesn't complete launching of the RV if python exits straight after.
+        # Adding a small sleep seems to allow it enough time to launch, although this is ugly.
+        # It was also found that launching Firefox or the Calculator app did work without the need for the sleep.
+        # So maybe it is something specific to RV?
+        # FIXME: We need to find a better fix than waiting and hoping that is long enough to allow RV to start launching.
+        time.sleep(0.5)
```

Comment:
This is ugly, and I'm not sure it's even necessary if you are running the app within engine other than shell. But as I was testing in the shell engine, the python process was ending straight after `Popen`, and that meant RV didn't launch in Python 3.
I did ask here on SO, on the off chance someone might have some ideas:
https://stackoverflow.com/questions/60092771/subprocess-popen-fails-to-execute-if-python-exits-straight-after

Reply:
Wow, that's a bummer. I suspect this is only going to be a problem for `tk-shell` because I'm assuming its our tank command process that's finishing before the subprocess is fully spawned and RV launches. That wouldn't be a concern in a DCC, where the main process isn't going to end after this code runs.

Reply:
Any way to test that? Maybe try in Max with Python 3 and if it's not a problem there then wrap the sleep in a condition to only run if we're in `tk-shell`?

Reply:
I've tested in Max, and there indeed doesn't seem to be a problem. So I've changed it so it only does this on the shell engine.